### DISCUSSION TITLE
What is the first 8 bytes of the instruction's data ?

### DISCUSSION THREAD

**Discussion by pooriagg** [otter-sec/anchor#2662] [Q&A] [answered]
In anchor programs, first 8 bytes of the account's data is the discriminator that fine but what is the first 8 bytes of the instruction's data when interaction with the anchor program's instruction ?! how we can compute that ?

**Comment by buffalojoec** [comment]
It's also an 8-byte discriminator!

It's calculated like this:

https://github.com/coral-xyz/anchor/blob/6cf200493a307c01487c7b492b4893e0d6f6cb23/lang/syn/src/codegen/program/dispatch.rs#L20-L26

So in other words, it's the first 8 bytes of the string `"global:{ix_name}"` where `ix_name` is the name of your instruction function.

**Reply by pooriagg** [reply]
thx for response, but my instruction's (method) name is "update" and the first 8 bytes of the data that calls that is [219, 200, 88, 176, 158, 63, 253, 127] can you just show a real example ?!

**Reply by buffalojoec** [reply, marked as answer]
```rust
let ix_name = "update";
let preimage = format!("global:{ix_name}");
let hash = solana_program::hash::hash(preimage.as_bytes()); // SHA256
let discriminator = &hash.to_bytes()[..8];
println!("discriminator: {:?}", discriminator);
```

**Reply by pooriagg** [reply]
thx a lot
